web-gateway/src/main/java/controllers/AbstractController.java

```diff
@@ -45,19 +48,14 @@ public AbstractController(MessagesApi messagesApi, UserService userService) {
     protected CompletionStage<Nav> loadNav(UUID userId) {
         return loadNav(Optional.of(userId));
     }
-
-    protected CompletionStage<Nav> loadNav(Optional<UUID> userId) {
-        return userService.getUsers(Optional.of(0), Optional.of(10)).invoke().thenApply(users -> {
-            Optional<User> currentUser = userId.flatMap(id -> {
-                for (User u: users.getItems()) {
-                    if (u.getId().equals(id)) {
-                        return Optional.of(u);
-                    }
-                }
-                return Optional.empty();
-            });
-            return new Nav(messagesApi.preferred(Collections.emptyList()), users.getItems(), currentUser);
-        });
-    }
-
-}
+    protected CompletionStage<Nav> loadNav(Optional<UUID> userId) { CompletionStage<Optional<User>> u;
+    if (userId.isPresent()) {
+        u = userService.getUser(userId.get())
+                .handleRequestHeader(authenticate(userId.get()))
+                .invoke()
+                .thenApply(resp -> Optional.of(resp)); } else {
+        u = CompletableFuture.completedFuture(Optional.empty()); }
+         CompletionStage<PaginatedSequence<User>> users = userService.getUsers(Optional.of(0), Optional.of(10)).invoke();
+    return users.thenCombineAsync( u, (userPaginatedSequence, maybeLoggedUser) -> new Nav(messagesApi.preferred(Collections.emptyList()), userPaginatedSequence.getItems(), maybeLoggedUser)); }
+
```

Comment:
I think it would also be helpful to refactor it to extract a `getUser` helper method like this:

```java
private CompletionStage<Optional<User>> getUser(userId) {
        if (userId.isPresent()) {
            return userService.getUser(userId.get())
                    .handleRequestHeader(authenticate(userId.get()))
                    .invoke()
                    .thenApply(resp -> Optional.of(resp));
        } else {
            return CompletableFuture.completedFuture(Optional.empty());
        }
}
```

You can then use it like this:

```java
CompletionStage<Optional<User>> u = getUser(userId);
```

It might also be good to rename `u` to something more descriptive, such as `currentUser`.

This makes it more clear what the intent of this method is, and separates out the details of how the current user is retrieved into a separate method.